### DISCUSSION TITLE
Code Coverage

### DISCUSSION THREAD

**Discussion by dwdraju** [icon-project/icon-bridge#138] [Ideas]
## Objective
To know how much of source code has been tested to give confidence on code that is being shifted to deployment

## Provider Options
1. [Coveralls](https://coveralls.io/)
2. [Codecov](https://about.codecov.io/)
3. [SonarQube](https://www.sonarqube.org/enterprise-edition/)

## Requirements
1. Support all the languages we use on [icon-bridge](https://github.com/icon-project/icon-bridge) repo
2. Reporting dashboard for the code coverage
3. Managed service and github(Github Actions) integration
4. Pricing: (Better if cost per repo than user??)

**Comment by andrii-kl** [comment]
1. Support all the languages we use on [icon-bridge](https://github.com/icon-project/icon-bridge) repo
  Java 
  Go
  Rust
  Solidity (smurt contracts)
  
I think Java and Go should have no problems with these freamworks. But I have doubts about solidity contracts. 
So need to research how it will be works for all of this languages.

**Reply by andrii-kl** [reply]
@dwdraju  Would you like to take this task on research?

**Reply by dwdraju** [reply]
Okay. Seems the Solidity test is doable through the npm package [solidity-coverage](https://www.npmjs.com/package/solidity-coverage). I was going through this medium post: https://medium.com/edgefund/testing-and-code-coverage-of-solidity-smart-contracts-660cb6291701

**Reply by andrii-kl** [reply]
I see. Does it mean that [Coveralls](https://coveralls.io/) can cover Java and Go, but for solidity we have to use another tool?

**Reply by bbist** [reply]
It will also include `rust` after near integration, which is already being added to `main` branch.

**Reply by bbist** [reply]
@dwdraju, @andrii-kl,
Our code would also include `rust`, so let's use something that has `rust` coverage as well.

**Comment by andrii-kl** [comment]
I would start from the [Coveralls](https://coveralls.io/).
If this one support all our languages, we can just start integration. 
We are going to make our repo open soon, so it will be free then.
